src/localstream.js

```diff
@@ -209,25 +241,43 @@ const localstream = (function() {
 		console.log('(ReachSDK::localstream::initAudioVideo)');
 		if (!initAudioVideoProgress) {
 			initAudioVideoProgress=true;
+			_getAllVideoSources();
 
 			navigator.getMedia = getUserMedia;
 			mLocalStreamAudioVideo = document.createElement('AUDIOVIDEO');
 			mLocalStreamAudioVideo.muted= true;
-			localAudioVideoStreams.push(mLocalStreamAudioVideo);
+			//localAudioVideoStreams.push(mLocalStreamAudioVideo);
+
+			let audioSource;
+			let videoSource;
+
+			if(videoSources[currentVideoSource]){
+				videoSource = videoSources[currentVideoSource].deviceId;
+			}
+
```

Comment:
The `exact` keyword makes the constraint required, guaranteeing it'll return only the right one, or fail. Thus you should add an error callback to `navigator.getMedia` to handle any possible rejection.